### DISCUSSION TITLE
Are Mini Dictionnaries in YAML shared accross the app

### DISCUSSION THREAD

**Discussion by nprime496** [kantord/LibreLingo#3373] [Q&A]
Hi, I see that it is possible to add a mini dictionnary for mapping of words.

Example : https://github.com/LibreLingo/LibreLingo/blob/main/courses/test-1/basics/skills/options-test-0.yaml

I want to know if these are shared across the app meaning that a translation added in some skill will be used in another.

**Comment by kantord** [comment]
They are shared across the same course. You are free to optionally define it in each skill, but it's only required to define it in one place. With the tooling that we have at the moment probably the best option you have if to rely on the errors to detect if you have a missing definition, and let that guide you in building the mini dictionary. But I guess we should have some sort of linting tool that removes duplicate definitions or perhaps even collect them into a global dictionary file

**Reply by Skivling** [reply]
A global dictionary file, perhaps separated by the module a word is introduced in, would be very nice. It would also add easy support for the Dictionary Notes someone discussed one time, as well as other things such as defining what type of word it is (noun, singular, masculine). Other things to be added could be URL links if one wants to link the words in their course to a reliable online dictionary (I want to do this with my Te Reo Māori course).
Linting tool would be great, avoid running and restarting a test server several times to find all the errors.
